recipe/meta.yaml

```diff
@@ -1,92 +1,100 @@
     - /usr/lib/libgcc_s.1.dylib
     - libc.so.6
 
-requirements:
-  run:
-    - {{ pin_subpackage(name.lower()+"_linux-64", exact=True) }}  # [linux]
-    - {{ pin_subpackage(name.lower()+"_osx-64", exact=True) }}    # [osx]
-    - {{ pin_subpackage(name.lower()+"_win-64", exact=True) }}    # [win]
-
-test:
-  commands:
-    - go help
-    - go version
-
 outputs:
-  - name: {{ name|lower }}-core
+  - name: {{ name }}                                                  # [cgo]
+    requirements:                                                     # [cgo]
+      run:                                                            # [cgo]
+        - {{ pin_subpackage(name+"_"+target_platform, exact=True) }}  # [cgo]
+    test:                                                             # [cgo]
+      commands:                                                       # [cgo]
+        - go version                                                  # [cgo]
+
+  - name: {{ name }}-{{ cgo_var }}
     script: build-core.sh   # [not win]
     script: build-core.bat  # [win]
     requirements:
       build:
-        {% if not win %}
-        - {{ compiler('c') }}
-        - {{ compiler('cxx') }}
-        - {{ compiler('fortran') }}
-        {% else %}
-        - {{ n }}toolchain
-        {% endif %}
-
-        - perl  # [not win]
+        - {{ compiler('go14') }}
+
+        - perl     # [not win]
+        - m2-sed   # [win]
+        - m2-grep  # [win]
+
+        - {{ compiler('c') }}              # [unix and cgo]
+        - {{ compiler('m2w64_c') }}        # [win and cgo]
+      run:                                 # [cgo]
+        - {{ compiler('c') }}              # [unix and cgo]
+        - {{ compiler('m2w64_fortran') }}  # [win and cgo]
     test:
       files:
         - hello.go
+        - hello-c.go  # [cgo]
       commands:
+        # Environment checks
         - go env
+        - test $(go env GOROOT) = ${CONDA_PREFIX}/go         # [unix]
+        - cmd /c if x%GOROOT% NEQ x%CONDA_PREFIX%\go exit 1  # [win]
+
+        - test $(go env CGO_ENABLED) -eq {{ cgo }}  # [unix]
+
+        # Test go can actually compile and run a basic program
         - go run hello.go
-        - cmd /c echo x%GOROOT% NEQ x%CONDA_PREFIX%\go        # [win]
-        - cmd /c if x%GOROOT% NEQ x%CONDA_PREFIX%\go exit 1   # [win]
 
-  - name: {{ name|lower }}_{{ conda_goplatform }}
-    script: build-platform.sh   # [not win]
-    script: build-platform.bat  # [win]
+        # Test cgo is working properly
+        - go run hello-c.go    # [cgo]
+
+  - name: {{ name }}_{{ target_platform }}         # [cgo]
+  - name: {{ name }}-nocgo_{{ conda_goplatform }}  # [not cgo]
+    script: build-{{ cgo_var }}-platform.sh   # [unix]
+    script: build-{{ cgo_var }}-platform.bat  # [win]
     requirements:
       build:
         - sed      # [osx]
         - m2-sed   # [win]
-        - m2-file  # [win]
-        - m2-grep  # [win]
-      host:
-        - {{ pin_subpackage("go-core", exact=True) }}
```

Comment:
This was taking unnecessary time. The build-time recipe is only installing proper (de)activate scripts.